### ISSUE TITLE
Help

### ISSUE THREAD

**Issue by EvelyneCalista** [MeijiTian/Spener#2]
Hi @MeijiTian ,

Thanks for your great work!
I am a graduate student and would like to reproduce you work. However, I face some issue and need you to clarify some things.

1.I would like to request more detailed information regarding the fan-beam geometry settings used in your study. Additionally, I am trying to locate the supplementary materials mentioned in the paper, but I have been unable to find them on the AAAI website.
Could you kindly let me know where I can access these materials?

2. For AAPM2016 dataset, do you use 1mm B30 or B45?

3. Since I am new to this study, Are the performance metrics (PSNR, SSIM, RMSE) calculated as averages over the entire dataset?

Thank you!

**Comment by MeijiTian** [comment]
Hello @EvelyneCalista，

Thanks for your interest in our work! 

1. I provide the supplementary materials here. You can find the geometry information in the materials or the config file in the code. 
2. For the AAPM 2016 dataset, we used the 1 mm B30 reconstructions.
3. Yes, the qualitative results shown in the table are the average of the test set. 

Please feel free to let me know if anything remains unclear or if you encounter other issues reproducing the results!

[Spener_SM.pdf](https://github.com/user-attachments/files/20272776/Spener_SM.pdf)

**Comment by EvelyneCalista** [comment]
Hi @MeijiTian ,

Thank you so much for your help and for providing the materials.
I have a few more questions—sorry for asking so many and thank you for your patience!

1. I noticed that the size of the input images differs from the raw data. May I ask if you resized the images before feeding them into the model? If so, could you kindly share which resizing method you used?
2. How did you split and select the slices for the training, validation, and test sets for the AAPM, CMB-CRC, and COVID datasets?
3. I observed that val_loader and test_loader are defined in train.py, but they don’t appear to be used during the training process. Could you please confirm if they serve any specific function?
4. May I confirm if the final output image is generated from the train_w_reg function, and whether the PSNR/SSIM metrics are calculated based on that image?
5. Should the SOD be set to -1 in the implementation, given that the supplementary states 363 while the config file uses 362?

**Comment by MeijiTian** [comment]
Hi @EvelyneCalista ,

1. We perform bilinear downsampling interpolation on the raw image with $512 \times 512$ to a size of $256 \times 256$. 
2. The training set and validation set are prepared for the supervised methods. It should be noted that our method is unsupervised. Therefore, like the traditional iterative reconstruction method, our method can be directly evaluated on the test set.  Thus, feel free to choose the slices for your validation and testing. 
3. The val_loader and test_loader do not perform any specific function. They help debug code in the development stage. 
4. Yes.
5. You can follow the setting in the config file since the forward process of generating sinograms is based on it.

**Comment by EvelyneCalista** [comment]
Sorry, I am still not quite sure regarding the source-to-detector distance, should it be 362 as specified in the config file, rather than 363 as mentioned in the supplementary material?

![Image](https://github.com/user-attachments/assets/85412135-431e-4b78-b3b5-2fbb4482dcda)


"fan_geometry":{
			"spacing" : 1.3,
			"det_count" : 613,
			**"SOD": 362**
		}

**Comment by MeijiTian** [comment]
Yeah, it should be 362 as shown in the config file. It is a typo in the supplementary material.  
You can also set it in a rational range to test different geometry settings.

**Comment by EvelyneCalista** [comment]
I see, thank you for your clarification!

Sorry, I have other questions:

1. May I ask why the input images are resized instead of using the original 512×512 resolution?
2. Regarding pre-processing and post-processing:

- Are the AAPM dataset images converted to Hounsfield Units (e.g., -1024 to 3072) and normalized (e.g., 0 to 1) as pre-processing step?
- When visualizing the CT images, should we apply a window of [-160, 240] HU, or is it acceptable to use the normalized [0, 1] range without further post-processing?
- Are similar steps applied to the COVID and CMB-CRC datasets?

3. I encountered some issues when working with the COVID-19 dataset settings:
- In train_w_sino, the variable D is computed as **D = int(img_size * (2**0.5))**. Could you clarify why this formula is used instead of assigning a fixed value from the config file? Since using this formula will encounter an error on reshape the pre_intensity.

![Image](https://github.com/user-attachments/assets/704e3e12-2ce1-418e-a045-df7d36635a88)

- In util.py, within the fanbeam_backprojection function, is it correct to modify the line to:
`x_tensor = torch.zeros((1, 1, 360, 360), dtype=torch.float32).to(device)`
specifically for the COVID-19 dataset?

**Comment by MeijiTian** [comment]
1. We use $256 \times 256$ image size to save computational memory and time. Most existing SVCT reconstruction methods also adopt this resolution.
2. 
+  We normalize the image from [-1024, 1024] HU to [0, 1] during preprocessing.
+  In the paper, we display the image in the range [0, 0.8], but you can choose a different window for visualization depending on which regions you want to highlight (e.g., soft tissue or bone).
+ Yes
3.  
+  $D$ is the number of sampling points per ray, which should be set in the config file. The value is defined as `D = int(img_size * sqrt(2))`, which is inherited from an earlier version of the code. For an image size of 256, this gives `D = int(256 * sqrt(2)) = 362`, roughly the diagonal length, to make sure the whole image is covered.
+ Yes.

**Comment by EvelyneCalista** [comment]
1. May I know why normalizing the image from [-1024, 1024] HU, not using [-1024,3072] (as min and max of HU value)?
2. How about for COVID19 data 360x360 image size? the formula D = int(360x sqrt(2)) = 509, and will get an error when reshaping the tensor

![Image](https://github.com/user-attachments/assets/5fe89dd7-ef22-43f6-a72a-3baa2228b18d)

**Comment by EvelyneCalista** [comment]
Sorry, I have one more question. I am confuse which final PSNR output reported in the paper. 
Does it come from **line 241** or **line 244** in the train.py? because each `train_w_sino`, `train_w_reg` also after the `bm3d `compute the metrics. and the metrics coming from **line 244** is lower than from **line 241**

![Image](https://github.com/user-attachments/assets/8f22de16-80c0-427f-bffe-fae2434e3882)
